# 5. Evaluation

The Evaluation phase assesses whether the modeling results meet the original business objectives. In CRISP-DM, the final results are not only the predictive models, but also the findings generated during the project.

RESULTS = MODELS + FINDINGS

## 5.1 Evaluate Results

### 5.1.1 Evaluation criteria from Business Understanding

The main objective of this project was to develop a tool capable of supporting better marketing decisions. From a business perspective, a useful model should help the company identify customers who are more likely to respond to future campaigns and reduce resources spent on customers with a very low probability of engagement.

For this reason, model evaluation was based not only on statistical performance, but also on its potential business value.

The following criteria were considered:

- **Ability to identify likely responders:** The model should successfully detect customers with a higher probability of accepting campaign offers. Correctly identifying these customers allows the Marketing Department to focus its efforts on individuals with greater conversion potential.

- **Reduction of inefficient targeting:** Contacting the entire customer base can generate unnecessary costs and reduce campaign efficiency. The model should help prioritize customers and reduce communication directed toward individuals unlikely to respond.

- **Support for marketing decisions:** In addition to generating predictions, the model should provide useful information for campaign planning and customer targeting strategies.

- **Balance between Recall and Precision:** Since the dataset showed strong class imbalance, particular attention was given to these metrics. Recall measures the model’s ability to detect customers who actually respond, while Precision measures how reliable those positive predictions are. A model with very high Recall but very low Precision may target many customers unnecessarily, while a model with high Precision but low Recall may miss valuable opportunities.

- **ROC-AUC:** ROC-AUC evaluates how well the model separates responders from non-responders across different probability thresholds. A higher value indicates stronger class discrimination and greater flexibility for future business decisions.

Since the dataset was imbalanced, Accuracy alone was not enough. The focus was on finding a model that could support useful marketing decisions while keeping a good balance across the main metrics.

### 5.1.2 Model comparison summary

In [5]:
# importing libraries
import pandas as pd

In [6]:
results_summary = pd.read_csv("../data/results_summary.csv")

results_summary

,Model,CV F1,Accuracy,Precision_1,Recall_1,F1_1,ROC_AUC,TN,FP,FN,TP
0,Tuned Gradient Boosting,0.597,0.895,0.732,0.455,0.561,0.902,370,11,36,30
1,Tuned Logistic Regression,0.565,0.826,0.445,0.742,0.557,0.901,320,61,17,49
2,Random Forest,NaN,0.850,0.494,0.606,0.544,0.858,340,41,26,40
3,Logistic Regression,NaN,0.855,0.507,0.545,0.526,0.871,346,35,30,36
4,Tuned Random Forest,0.591,0.861,0.531,0.515,0.523,0.879,351,30,32,34
5,Gradient Boosting,NaN,0.893,0.781,0.379,0.510,0.897,374,7,41,25
6,Tuned Decision Tree,0.519,0.852,0.500,0.409,0.450,0.778,354,27,39,27
7,Decision Tree,NaN,0.761,0.328,0.591,0.422,0.758,301,80,27,39


#### Key Findings

The models produced different results across the evaluation metrics.

**Tuned Gradient Boosting** achieved the highest test-set F1-score and ROC-AUC:

* Accuracy: **0.90**
* Precision: **0.73**
* Recall: **0.45**
* F1-score: **0.56**
* ROC-AUC: **0.90**

The model correctly identified **30 responders**, produced **11 false positives**, and missed **36 responders**.

**Tuned Logistic Regression** achieved the highest Recall at **0.74**, correctly identifying **49 responders**. However, it also produced **61 false positives**.

**Gradient Boosting** achieved the highest Precision at **0.78**, but its Recall was lower at **0.38**.

**Random Forest** achieved Recall of **0.61** and an F1-score of **0.54**.

The Decision Tree models produced the lowest F1-scores among the tested approaches.

### Final Model Selection: Tuned Gradient Boosting

### 5.1.4 Business Interpretation of the Selected Model

The selected Tuned Gradient Boosting model:

* correctly identified **30 customers** who responded to the campaign;
* missed **36 customers** who responded;
* incorrectly selected **11 customers** who did not respond;
* correctly excluded **370 customers** who did not respond.

The model classified 41 customers as likely responders, of whom 30 actually responded.

These results show that the model limited the number of unnecessary positive predictions while identifying a portion of the responder group.


### 5.1.5. Findings Beyond the Model

The analysis identified several customer characteristics associated with campaign response:

* customers with more recent purchases tended to show higher response rates;
* previous campaign acceptance was associated with future response;
* higher customer spending was associated with greater campaign engagement;
* customers with more catalogue and web purchases tended to respond more frequently;
* household composition showed differences in response behaviour.

These findings are based on patterns observed in the available dataset and do not establish causal relationships.



## 5.2. Process Review

## 5.2 Process Review

The project followed the main CRISP-DM phases: Business Understanding, Data Understanding, Data Preparation, Modeling, Evaluation, and Deployment planning.

Data-quality issues were identified and corrected before modeling. The variables `Age`, `Seniority`, `Total_Spending`, and `Accepted_Total` were created to represent customer characteristics and behaviour.

Missing values were handled after the train-test split. Scaling was applied to Logistic Regression, while tree-based models used non-scaled data. SMOTE was applied only to the training data used for the baseline models. Four classification algorithms were tested, and hyperparameter tuning was performed using GridSearchCV. Tuning improved some models but did not improve every evaluation metric in all cases.

The main limitation was the absence of information about the actual cost of contacting a customer and the value of a successful response. Therefore, model selection was based only on statistical metrics. The models were also evaluated using one historical train-test split. Performance may differ when the model is applied to future campaigns or new customer data.

## 5.3 Determine Next Steps

## 5.3 Determine Next Steps

Based on the evaluation results, the project can proceed to the Deployment phase using **Tuned Gradient Boosting** as the selected model.

The model achieved the highest test-set F1-score and ROC-AUC while maintaining Precision of **0.73**.

The next steps are:

* prepare the same preprocessing steps for new customer data;
* generate response probabilities for future campaigns;
* define a probability threshold for selecting campaign recipients;
* compare predictions with actual campaign responses;
* monitor Precision, Recall, F1-score, and ROC-AUC over time;
* retrain the model when new campaign data becomes available.

Future work could incorporate campaign costs and customer value so that the classification threshold can be selected according to expected campaign profitability.
